validations/src/validations.rs

```diff
@@ -2060,15 +2072,26 @@ pub fn validate_block(
         // with the genesis_block_hash
         validate_genesis_block(block, consensus_constants.genesis_hash).map_err(Into::into)
     } else {
-        let total_identities = u32::try_from(rep_eng.ars().active_identities_number())?;
-        let (target_hash, _) = calculate_randpoe_threshold(
-            total_identities,
-            consensus_constants.mining_backup_factor,
-            block_epoch,
-            consensus_constants.minimum_difficulty,
-            consensus_constants.epochs_with_minimum_difficulty,
-            active_wips,
-        );
+        let target_hash;
+        if protocol_version == ProtocolVersion::V2_0 {
+            let validator = block.block_sig.public_key.pkh();
+            let validator_key = StakeKey::from((validator, validator));
+            let eligibility = stakes.mining_eligibility(validator_key, block_epoch);
+            if eligibility == Ok(Eligible::No(InsufficientPower)) || eligibility == Ok(Eligible::No(NotStaking)) {
+                return Err(BlockError::ValidatorNotEligible{ validator }.into());
+            }
+            target_hash = Hash::max();
+        } else {
+            let total_identities = u32::try_from(rep_eng.ars().active_identities_number())?;
+            (target_hash, _) = calculate_randpoe_threshold(
+                total_identities,
+                consensus_constants.mining_backup_factor,
+                block_epoch,
+                consensus_constants.minimum_difficulty,
+                consensus_constants.epochs_with_minimum_difficulty,
+                active_wips,
+            );
+        }
```

Comment:
For uniformity with our general code style:
```suggestion
        let target_hash = if protocol_version == ProtocolVersion::V2_0 {
            let validator = block.block_sig.public_key.pkh();
            let validator_key = StakeKey::from((validator, validator));
            let eligibility = stakes.mining_eligibility(validator_key, block_epoch);
            if eligibility == Ok(Eligible::No(InsufficientPower)) || eligibility == Ok(Eligible::No(NotStaking)) {
                return Err(BlockError::ValidatorNotEligible{ validator }.into());
            }
            
            Hash::max()
        } else {
            let total_identities = u32::try_from(rep_eng.ars().active_identities_number())?;
            (target_hash, _) = calculate_randpoe_threshold(
                total_identities,
                consensus_constants.mining_backup_factor,
                block_epoch,
                consensus_constants.minimum_difficulty,
                consensus_constants.epochs_with_minimum_difficulty,
                active_wips,
            );
            
            target_hash
        }
```